independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/ClassPathResourceIndex.java

```diff
@@ -239,19 +240,19 @@ public void addResourceMapping(ClassPathElement classPathElement, String resourc
             }
         }
 
-        public void addParentFirstResource(ClassPathElement classPathElement, String resource) {
+        public void addParentFirstResource(String resource) {
             parentFirstResources.add(resource);
         }
 
-        public void addBannedResource(ClassPathElement classPathElement, String resource) {
+        public void addBannedResource(String resource) {
             bannedResources.add(resource);
         }
 
         public ClassPathResourceIndex build() {
             Map<String, ClassPathElement[]> compactedResourceMapping = new HashMap<>(resourceMapping.size());
             for (Entry<String, List<ClassPathElement>> resourceMappingEntry : resourceMapping.entrySet()) {
                 compactedResourceMapping.put(resourceMappingEntry.getKey(),
-                        resourceMappingEntry.getValue().toArray(new ClassPathElement[resourceMappingEntry.getValue().size()]));
+                        resourceMappingEntry.getValue().toArray(EMPTY_CLASSPATH_ELEMENT));
```

Comment:
Do you have a pointer for this? I remembered I was told your thing was better now but I had a look at the code of the JDK the other day and it wasn't that obvious.

Not asking to justify but if you have a pointer handy, I'll be interested in refreshing my memory.

Reply:
https://shipilev.net/blog/2016/arrays-wisdom-ancients/ is why I use the idiom.